src/Database/PostgreSQL/PQTypes/Internal/Query.hs

```diff
@@ -30,11 +35,26 @@ runQueryIO sql st = do
     -- thread for the query execution and while we wait for its completion, we
     -- are able to receive asynchronous exceptions (assuming that threaded GHC
     -- runtime system is used) and react appropriately.
-    queryRunner <- async . restore $ do
-      let allocParam = ParamAllocator $ withPGparam cdPtr
-      (paramCount, res) <- withSQL sql allocParam $ \param query -> (,)
-        <$> (fromIntegral <$> c_PQparamCount param)
-        <*> c_PQparamExec cdPtr nullPtr param query c_RESULT_BINARY
+    let allocParam = ParamAllocator $ withPGparam cdPtr
+    queryRunner <- async . restore $ withSQL sql allocParam $ \param query format -> do
+      let queryName = BS.concat
```

Comment:
@marco44 does that satisfy you? I wanted to make it longer, but it looks like PostgreSQL only considers first 64 characters :thinking:

Reply:
Never mind, the trick is to put the hash first.

Reply:
From what I see, you're trying to remove all the current mechanism to replace it with prepared queries all over the place, right ? I think it's overkill, and could even be dangerous, we could anyway still need the old mechanism as a fallback: prepared queries static plan is not something we'd want all the time. I was more thinking of simply being able to use this new mechanism for a few queries (like the top few that are executed a lot with a small runtime and we're sure have a static plan, like fetching a document). For seldom run queries, prepared queries can be worse than normal one (that's why we have the 2 mechanisms :) ) : you need more round trips on the network, and you'll be compiling 2 plans: the static one, and the one taking into account all the parameters, for the first few runs.

To answer the original question, the "name" type is 64 chars (that's a compile option, but the default is 64). So yes, 64 is good. But why not pass it a name as a parameter in the function call ? That would be much easier for me when looking at the logs (I'll probably see a bunch of EXECUTE this_string). In my plan, we were going to do this for a select few queries, those where execution is small compared to planning, for optimization, not remove all that is in place